src/core/types.ts

```diff
@@ -24,10 +36,30 @@ export interface Structure extends Structure.Tranformable<(data: any) => any> {
   _variant?: Partial<Structure.Variant>;
 }
 
-export type GlobalMixin = TagMixin & {
-  shouldUpdate?: boolean | ((data: object, nextOpts: object) => boolean);
-};
+/**
+ * The type of the `shouldUpdate` function.
```

Comment:
Describe what the `shouldUpdate` function does or just reference Riot?

eg, `Runs before every component update and determines whether to trigger an update`